### PR TITLE
Update Nektar++ package

### PR DESCRIPTION
**Pull request by JamesEdgeley**

### CODE REVIEW

**Comment by JamesEdgeley** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/nektar/package.py [hunk 1]
```diff
@@ -3,37 +3,117 @@
...
 class Nektar(CMakePackage):
     """Nektar++: Spectral/hp Element Framework"""
 
     homepage = "https://www.nektar.info/"
-    url = "https://gitlab.nektar.info/nektar/nektar/-/archive/v4.4.1/nektar-v4.4.1.tar.bz2"
+    git = "https://gitlab.nektar.info/nektar/nektar.git"
 
-    version("5.0.0", sha256="5c594453fbfaa433f732a55405da9bba27d4a00c32d7b9d7515767925fb4a818")
-    version("4.4.1", sha256="71cfd93d848a751ae9ae5e5ba336cee4b4827d4abcd56f6b8dc5c460ed6b738c")
+    version("5.4.0", commit="4365d5d7156139f238db962deae5eb25e0437d12", preferred=True)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/nektar/package.py]: I think you meant to name this `5.5.0` (which matches the commit in the repo).

File: var/spack/repos/builtin/packages/nektar/package.py [hunk 2]
```diff
@@ -3,37 +3,117 @@
...
+        "vortexwave_solver",
+        default=False,
+        description="Builds an executable associated with the Vortex Wave solver",
+    )
 
     depends_on("cmake@2.8.8:", type="build", when="~hdf5")
     depends_on("cmake@3.2:", type="build", when="+hdf5")
 
     depends_on("blas")
+    depends_on("zlib")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/nektar/package.py:102]: Des this dependency apply to older versions or for some (perhaps newer) version on?

(See https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-dependencies)

File: var/spack/repos/builtin/packages/nektar/package.py [hunk 3]
```diff
@@ -50,10 +132,60 @@ def cmake_args(self):
...
         args.append("-DNEKTAR_USE_ARPACK=%s" % hasfeature("+arpack"))
+        args.append("-DNEKTAR_USE_FFTW=%s" % hasfeature("+fftw"))
         args.append("-DNEKTAR_USE_HDF5=%s" % hasfeature("+hdf5"))
-        args.append("-DNEKTAR_USE_SCOTCH=%s" % hasfeature("+scotch"))
+        args.append("-DNEKTAR_USE_MKL=%s" % hasfeature("^intel-oneapi-mkl"))
+        args.append("-DNEKTAR_USE_MPI=%s" % hasfeature("+mpi"))
+        args.append("-DNEKTAR_USE_OPENBLAS=%s" % hasfeature("^openblas"))
         args.append("-DNEKTAR_USE_PETSC=OFF")
+        args.append("-DNEKTAR_USE_SCOTCH=%s" % hasfeature("+scotch"))
+        args.append("-DNEKTAR_USE_THREAD_SAFETY=ON")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/nektar/package.py]: The `CMakePackage` has some useful helpers you might want to consider adopting, like `self.define_from_variant()`.  See the docs at https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake.

We are also encouraging people to switch to Python f-strings (see https://docs.python.org/3/tutorial/inputoutput.html).  We are converting the Spack core as we work on issues and features.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed all of the new commits exist but one of the version directives is specifying the wrong version (name).  I have a question and a suggestion.

File: var/spack/repos/builtin/packages/nektar/package.py (hunk 2 above)
Comment (JamesEdgeley) [reply var/spack/repos/builtin/packages/nektar/package.py:102]: As far as I know all versions require zlib, although nektar will try and build it itself if it doesn't detect it on the system.

File: var/spack/repos/builtin/packages/nektar/package.py (hunk 3 above)
Comment (JamesEdgeley) [reply var/spack/repos/builtin/packages/nektar/package.py]: Thanks for the suggestions

**Comment by JamesEdgeley** [comment]
@spackbot fix style

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**